rollup/tracing/tracing.go

```diff
@@ -192,6 +197,41 @@ func CreateTraceEnv(chainConfig *params.ChainConfig, chainContext core.ChainCont
 	return env, nil
 }
 
+func (env *TraceEnv) ResetForPartialTrace(partialBlk *types.Block) error {
+
+	if env.StorageTrace == nil {
+		return fmt.Errorf("not init")
+	}
+
+	// TODO: can we chained the RootBefore / After?
+	oldStorage := env.StorageTrace
+
+	// only reset which can be reset
+	env.signer = types.MakeSigner(env.chainConfig, partialBlk.Number())
+	env.StorageTrace = &types.StorageTrace{
+		RootBefore:     oldStorage.RootBefore,
+		RootAfter:      partialBlk.Root(),
```

Comment:
is this line O(1) read root or commit zktrie? `RootAfter:      partialBlk.Root(),`